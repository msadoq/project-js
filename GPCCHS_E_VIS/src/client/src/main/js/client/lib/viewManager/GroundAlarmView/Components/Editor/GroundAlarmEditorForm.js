import React, { PropTypes, Component } from 'react';
import classnames from 'classnames';
import { Form } from 'react-bootstrap';
import { Field, reduxForm } from 'redux-form';
import HorizontalFormGroup from '../../../../windowProcess/commonReduxForm/HorizontalFormGroup';
import ClearSubmitButtons from '../../../../windowProcess/commonReduxForm/ClearSubmitButtons';
import ReactSelectField from '../../../../windowProcess/commonReduxForm/ReactSelectField';
import * as constants from '../../../../constants';

const MODES = [
  { value: constants.GMA_ALARM_MODE_ALL, label: 'All' },
  { value: constants.GMA_ALARM_MODE_NONNOMINAL, label: 'Non nominal' },
  { value: constants.GMA_ALARM_MODE_TOACKNOWLEDGE, label: 'To Acknowledge' },
];

class EntryPointConnectedData extends Component {
  componentDidMount() {
    setTimeout(this.props.reset, 0);
  }

  render() {
    const {
      handleSubmit,
      pristine,
      reset,
      submitting,
      valid,
      timelines,
      domains,
    } = this.props;

    return (
      <Form
        horizontal
        onSubmit={handleSubmit}
        className={classnames(
          { 'redux-form-dirty': !pristine },
          'redux-form-padded'
        )}
      >
        <ClearSubmitButtons
          pristine={pristine}
          submitting={submitting}
          reset={reset}
          valid={valid}
        />
        <br />
        <HorizontalFormGroup label="Domain">
          <Field
            name="domain"
            clearable={false}
            component={ReactSelectField}
            options={domains.map(d =>
              ({
                label: d.name,
                value: d.name,
              })
            ).concat({
              label: '*',
              value: '*',
            })}
          />
        </HorizontalFormGroup>

        <HorizontalFormGroup label="Timeline">
          <Field
            name="timeline"
            clearable={false}
            component={ReactSelectField}
            options={timelines.map(t =>
              ({
                label: t.id,
                value: t.id,
              })
            ).concat({
              label: '*',
              value: '*',
            })}
          />
        </HorizontalFormGroup>

        <HorizontalFormGroup label="Mode">
          <Field
            name="mode"
            clearable={false}
            component={ReactSelectField}
            options={MODES}
          />
        </HorizontalFormGroup>
      </Form>
    );
  }
}

EntryPointConnectedData.propTypes = {
  // eslint-disable-next-line react/no-unused-prop-types, "DV6 TBC_CNES Supported by ReduxForm HOC"
  initialValues: PropTypes.shape({
    formula: PropTypes.string,
    domain: PropTypes.string,
    timeline: PropTypes.string,
  }).isRequired,
  timelines: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string,
  })).isRequired,
  domains: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  reset: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired,
  valid: PropTypes.bool.isRequired,
};

const requiredFields = [/* 'formula', 'domain', 'timeline' */];
const validate = (values = {}) => {
  const errors = {};

  requiredFields.forEach((field) => {
    if (!values[field]) {
      errors[field] = 'Required';
    }
  });
  return errors;
};

export default reduxForm({
  validate,
  enableReinitialize: true,
})(EntryPointConnectedData);
