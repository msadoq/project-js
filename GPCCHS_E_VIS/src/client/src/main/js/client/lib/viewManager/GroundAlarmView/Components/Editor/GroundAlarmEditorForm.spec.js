import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import GroundAlarmEditorForm from './GroundAlarmEditorForm';

const propsStub = {
  initialValues: {},
  handleSubmit: () => null,
  pristine: true,
  reset: () => null,
  submitting: true,
  valid: true,
};

describe('GroundAlarmEditorForm :: render', () => {
  test('GroundAlarmEditorForm :: render', () => {
    const tree = shallow(
      <GroundAlarmEditorForm
        {...propsStub}
      />
    );
    expect(toJson(tree)).toMatchSnapshot();
  });
});
