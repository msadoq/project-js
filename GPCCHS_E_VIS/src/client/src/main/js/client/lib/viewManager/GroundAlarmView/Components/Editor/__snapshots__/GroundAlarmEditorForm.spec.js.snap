// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroundAlarmEditorForm :: render snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GroundAlarmEditorForm
    handleSubmit={[Function]}
    initialValues={Object {}}
    pristine={true}
    reset={[Function]}
    submitting={true}
    valid={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "handleSubmit": [Function],
      "initialValues": Object {},
      "pristine": true,
      "reset": [Function],
      "submitting": true,
      "valid": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "handleSubmit": [Function],
        "initialValues": Object {},
        "pristine": true,
        "reset": [Function],
        "submitting": true,
        "valid": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
